### ISSUE TITLE
Error during Refund

### ISSUE THREAD

**Issue by RomanovM** [nopSolutions/Authorize.Net-plugin-for-nopcommerce#21]
Authorize/Capture works as expected, however Refund throws an error about the expirationDate exceeding maxLength.
The plugin code matches their code sample, and "XXXX" is sent [instead of the real expirationDate](https://github.com/nopSolutions/Authorize.Net-plugin-for-nopcommerce/blob/nopcommerce-4.30/Nop.Plugin.Payments.AuthorizeNet/AuthorizeNetPaymentProcessor.cs#L521)

```
var creditCard = new creditCardType
{
    cardNumber = lastFourDigitsCardNumber,
     expirationDate = "XXXX"
};
```

It worked fine in Dev/Sandbox, but fails in Prod.  
Plugin API credentials are correct.  
Nothing in the plugin code modifies the expirationDate after it is set to "XXXX", and it's unclear how "XXXXXXXXXXXXX..." is being sent, but maxLength == 4:

```
expirationDate   Required.
The card's expiration date, masked for refunds.
  
String, four characters.

For refunds, use XXXX instead of the card expiration date
```

Source: https://www.nopcommerce.com/en/boards/topic/87096/authorizenet-plugin-error-during-refund

**Comment by ted-phillips** [comment]
Please disregard, I finally traced the error to some customization and not your code.

**Comment by skoshelev** [comment]
Hi @ted-phillips. Thank you for telling us

Closed #21
